adal/adal-rules.pro

```diff
@@ -0,0 +1,110 @@
+# Add project specific ProGuard rules here.
+# By default, the flags in this file are appended to flags specified
+# in C:\Program Files\Android\android-sdk/tools/proguard/proguard-android.txt
+# You can edit the include path and order by changing the proguardFiles
+# directive in build.gradle.
+#
+# For more details, see
+#   http://developer.android.com/guide/developing/tools/proguard.html
+
+# Add any project specific keep options here:
+
+# If your project uses WebView with JS, uncomment the following
+# and specify the fully qualified class name to the JavaScript interface
+# class:
+#-keepclassmembers class fqcn.of.javascript.interface.for.webview {
+#   public *;
+#}
+
+-dontskipnonpubliclibraryclasses
+
+##---------------Begin: proguard configuration for Common  --------
+-keep,includedescriptorclasses class com.microsoft.identity.common.exception.** { *; }
+-keep,includedescriptorclasses class com.microsoft.identity.common.internal.broker.** { *; }
+-keep,includedescriptorclasses class com.microsoft.identity.common.internal.cache.** { *; }
+-keep,includedescriptorclasses class com.microsoft.identity.logging.internal.dto.** { *; }
+-keep,includedescriptorclasses class com.microsoft.identity.internal.logging.** { *; }
+-keep,includedescriptorclasses class com.microsoft.identity.common.internal.providers.** { *; }
+-keep,includedescriptorclasses class com.microsort.identity.common.internal.telemtery.** { *; }
+-keep,includedescriptorclasses class com.microsoft.identity.common.internal.ui.** { *; }
+-dontwarn com.microsoft.identity.common.internal.ui.**
+-keep,includedescriptorclasses class com.microsoft.device.display.** { *; }
+-keep,includedescriptorclasses class org.apache.harmony.xnet.provider.jsse.NativeCrypto { *; }
+
+##---------------Begin: proguard configuration for crypto classes  --------
```

Comment:
Where are these classes coming from? I don't see us shipping these -- are they transitive dependencies?

Reply:
The *display one is so that nobody removes the dual-screen code, I believe.  The other one is misplaced.  I'm moving them and will update this PR with methodology.